### ISSUE TITLE
qualityPhredScores Error

### ISSUE THREAD

**Issue by cgcole** [roblanf/sangeranalyseR#76]
Hello,

I am running into an issue with using SangerAlignment() on some of my .ab1 files. I have used it in the past with no problem, but for files from my last two batches of sequencing, I run into the following error code: 

Error in .local(.Object, ...) : 
  'qualityPhredScores' length cannot be zero.PARAMETER_VALUE_ERROR
In addition: Warning message:
In read.abif(readFileName) : unimplemented legacy type found in file

I can't seem to identify the difference between the .ab1 files that is triggering the error.

Thanks!
Cody

**Comment by Kuanhao-Chao** [comment]
Hi @CodyG12, thanks for the report. The `unimplemented legacy type found in file` warning was the giveaway — `sangerseqR::read.abif` was succeeding on those files but returning an empty `PCON.2` quality block, which then failed our hard `'qualityPhredScores' length cannot be zero` validator. We've seen this pattern from older Beckman / 3500 firmware ABIFs.

**Fix (Phase 15)**: `SangerRead.initialize` now detects the missing/empty `PCON.2` block, synthesises a flat Phred-30 quality vector (one entry per detected peak), and logs a `MISSING_QUALITY_SCORES_WARN` so it's clear the trimmed sequence's quality assessment is not based on the original instrument's per-base scores. The rest of the pipeline (basecalling, alignment, consensus) runs normally.

Caveats: with synthetic flat quality, M1/M2 trimming becomes a no-op (no quality differential to trim against). The `SangerRead@QualityReport` slot is still populated and consistent so Shiny / report rendering continue to work, but you should manually inspect the consensus from these reads.

Two regression tests:

- Real ABIF with intact `PCON.2` continues to use the instrument quality scores (no behaviour change for the happy path).
- Doctored ABIF with empty `PCON.2` falls through to the synthesised Phred-30 path; assertion: every score in the resulting `@QualityReport@qualityPhredScores` slot equals 30.

Available in the next Bioconductor `devel` build. Install from GitHub for immediate testing:

```r
remotes::install_github("roblanf/sangeranalyseR", ref = "devel")
```

Closing as fixed.
